### DISCUSSION TITLE
Garden Embed Federation

### DISCUSSION THREAD

**Discussion by uroybd** [oleeskild/obsidian-digital-garden#178] [Ideas]
I'm toying with an idea of "federated garden embed" which will allow users to cross embed contents. The project is ambitious. But, implemented properly can do wonder. Imagine quoting other people over the internet but treating them like native content.

I'll update on it here: https://hermitage.utsob.me/entities/concept/obsidian-federated-embed/

But, here's the basic idea:

## Abstract
Digital Gardens are great for nurturing thought. And, evergreen notes expands but seldom scraped altogether. Therefore, Ideally, to build **campfire** of multiple gardener, cross-embedding between vaults/gardens.

### Publisher
The publisher, be it a plugin like Obsidian Digital Garden, or a dedicated one will generate a manifest file, let's say, `obsidian-federation.json` to the root of the public URL with this format:

```json
{
  "version": "1.0.0",
  "last_updated": "<unix timestamp>",
  "content_root": "content root to prepend before content_url",
  "published_root": "published root to prepend before published_url",
  "notes": [
    {
      "title": "Some Title",
      "aliases": ["An alias"],
      "content_url": "relative url accessible from web to get the raw content",
      "published_url": "optional relative publicly available url",
      "last_updated": "<unix timestamp>"
    }
  ]
}
```

### Consumer
Consumer plugin will leverage existing callout syntax and provide auto-complete to help the user.

```md
> [!FED] URL/path
> <pulled_content>
>
> updated_at: <timestamp>
```

The plugin will get the manifest file based on the `URL`, then load the contents and suggest based on title, alias, and content URL then append the text as quote.

For partial embed, it will support block and header embed like existing syntax. There are also two new syntax for partial embed:

```
URL/path#:1-3 for line based embed
URL/path#>p:3 for tag-based embed, paragraph 3 in this case.
```

**Comment by andrey-jef** [comment]
Great idea. It is similar to what Dendron imagined to implement. Though they decided to move their project into maintenance mode due to product-market-fit failure.

**Comment by oleeskild** [comment]
This would be really interesting to try to implement. Have you given any thought to how versioning should work? Given that you are linking to other people's writings, they are subject to change at any time.
Should each content entry have a list of all versions? Should we snapshot the content as it was when the embed was created? Should we not take this into account at all? Any ideas?

**Reply by uroybd** [reply]
I thought, we should copy the relevant part and timestamp them. This is important for coherence, historicity etc. Plugin can then handle version management on subsequent pulls and ask users whether to overwrite new content.
